src/plugins/saved_objects_management/public/management_section/objects_table/saved_objects_table.tsx

```diff
@@ -178,43 +179,57 @@ export class SavedObjectsTable extends Component<SavedObjectsTableProps, SavedOb
   }
 
   fetchCounts = async () => {
-    const { allowedTypes } = this.props;
+    const { allowedTypes, namespaceRegistry } = this.props;
     const { queryText, visibleTypes, visibleNamespaces } = parseQuery(this.state.activeQuery);
 
-    const filteredTypes = allowedTypes.filter(
-      (type) => !visibleTypes || visibleTypes.includes(type)
-    );
+    const filteredTypes = filterQuery(allowedTypes, visibleTypes);
+
+    const availableNamespaces = namespaceRegistry.getAll()?.map((ns) => ns.id) || [];
+
+    const filteredCountOptions = {
+      typesToInclude: filteredTypes,
+      searchString: queryText,
+    };
+
+    if (availableNamespaces?.length) {
+      const filteredNamespaces = filterQuery(availableNamespaces, visibleNamespaces);
+      filteredCountOptions.namespacesToInclude = filteredNamespaces;
+    }
 
     // These are the saved objects visible in the table.
     const filteredSavedObjectCounts = await getSavedObjectCounts(
       this.props.http,
-      filteredTypes,
-      visibleNamespaces,
-      queryText
+      filteredCountOptions
     );
 
     const exportAllOptions: ExportAllOption[] = [];
     const exportAllSelectedOptions: Record<string, boolean> = {};
 
-    Object.keys(filteredSavedObjectCounts).forEach((id) => {
+    const filteredTypeCounts = filteredSavedObjectCounts.type || {};
+
+    Object.keys(filteredTypeCounts).forEach((id) => {
       // Add this type as a bulk-export option.
       exportAllOptions.push({
         id,
-        label: `${id} (${filteredSavedObjectCounts[id] || 0})`,
+        label: `${id} (${filteredTypeCounts[id] || 0})`,
       });
 
       // Select it by default.
       exportAllSelectedOptions[id] = true;
     });
 
+    const countOptions = {
+      typesToInclude: allowedTypes,
+      searchString: queryText,
+    };
+
+    if (availableNamespaces?.length) {
```

Comment:
Based on line 187, `availableNamespaces` is always an array:
```suggestion
    if (availableNamespaces.length) {
```

Reply:
Changed